### DISCUSSION TITLE
how to data statistics.

### DISCUSSION THREAD

**Discussion by zjc1750514326** [apache/skywalking#12684] [Q&A] [answered]
i use skywalking in my project. now i have many trace name getBalance.this trace request  interface /customer/getCustomerBalance  in other application which do not report data. now i want to get /customer/getCustomerBalance spend time statistics.but mark url has show all trace. i just focus on /customer/getCustomerBalance. point did not show /customer/getCustomerBalance.like this picture
![image](https://github.com/user-attachments/assets/7d45fc9d-252a-4808-b4e1-b2c6419cd594)

**Comment by wu-sheng** [comment]
I am not following your asking.

What is that URL meaning? If it is a URI, it should be an endpoint, doesn't it?

Why do you show me the trace list?

**Reply by wu-sheng** [reply]
More strange is, you use that as a tag? URL should not be tag, how do you set up all these?

**Reply by zjc1750514326** [reply]
My application has requested third-party interfaces/customer/getCustomer Balance many times. I hope to directly obtain the time consumption of requesting third-party interfaces in the Skywalk interface, such as p99. However, when I mark the search, the entire link appears, but I am not concerned with the entire link. I only care about the situation of requesting third-party interfaces. Unable to search third-party interfaces at endpoints

**Reply by wu-sheng** [reply, marked as answer]
Client side calling is not the endpoint. So, no matter you are doing at the OAP side, would not change that.

You need to follow https://skywalking.apache.org/docs/skywalking-java/latest/en/setup/service-agent/java-agent/logic-endpoint/ to setup logic endpoint.
Then, you can see it as same as other endpoint statistics.

**Reply by zjc1750514326** [reply]
![image](https://github.com/user-attachments/assets/7f87fad7-8393-48a9-aa1a-6aa8971931a0)
Does this mean that I have so many trace: GET:/order/balance requests for/customer/getCustomer Balance that I can only click on them one by one to view, rather than just checking the time consumption for/customer/getCustomer Balance?

**Reply by wu-sheng** [reply]
Do you understand what is a logic endpoint and what is an endpoint? 

I think I said it very clear. You need to build this logic endpoint, then you have the statistics. I am feeling you don't understand these two concepts.

**Reply by wu-sheng** [reply]
Trace page has nothing to do with the statistics. It is never related.

**Reply by zjc1750514326** [reply]
> Do you understand what is a logic endpoint and what is an endpoint?
> 
> I think I said it very clear. You need to build this logic endpoint, then you have the statistics. I am feeling you don't understand these two concepts.



Do you use @Trace to manually dot the endpoint on the method for statistical purposes？

**Reply by wu-sheng** [reply]
I don't use anything. The doc has shared how to do logic endpoint. Use annotation/toolkit/plugins, that is your choice.
